snyk/cli
# chore: Added support for custom rules [CFG-1830]

#### What does this PR do?

- Adds support for custom rules in the issue description for the new test output, by:
    - Rather than a docs URL, which custom rules don't have, we display the custom rule's public ID.

#### Where should the reviewer start?

    src/lib/formatters/iac-output/v2/issues-list/issue.ts

#### How should this be manually tested?

1. Run the `snyk iac test` command with a custom rules bundles.
2. Ensure that all the expected issues generated from custom rules are displayed, and have the following format for the `Rule` property: `Rule: custom rule: <rule-public-ID>`

#### Any background context you want to provide?

Currently, for issues generated by custom rules, we do not show the Rule property in the issue description, as it normally holds the documentation URL and custom rules do not have it. Since we use these documentation URLs to display security rule public IDs, we don’t include the custom rule public IDs in the output at all. 

We’d like to update the Rule property to specify the rule’s public ID, in the case of custom rules.

#### What are the relevant tickets?

- [CFG-1830](https://snyksec.atlassian.net/browse/CFG-1830)

#### Screenshots

![image](https://user-images.githubusercontent.com/46415136/167114690-90489c8a-97c2-4e67-bf90-733920de1c1d.png)

#### Additional notes

- The design used for the `Rule` property in this PR was approved in [the following Slack thread](https://snyk.slack.com/archives/C030UA3U1RB/p1650465503629659).
